2a_model/src/models/Snakefile_base.smk

```diff
@@ -0,0 +1,208 @@
+import os
+import tensorflow as tf
+import numpy as np
+import pandas as pd
+
+from river_dl.preproc_utils import asRunConfig
+from river_dl.preproc_utils import prep_all_data
+from river_dl.evaluate import combined_metrics
+from river_dl.postproc_utils import plot_obs, plot_ts
+from river_dl.predict import predict_from_arbitrary_data
+from river_dl.train import train_model
+from river_dl import loss_functions as lf
+
+out_dir = os.path.join(config['out_dir'], config['exp_name'])
+loss_function = lf.multitask_rmse(config['lambdas'])
+
+
+rule as_run_config:
+    output:
+        "{outdir}/asRunConfig.yml"
+    run:
+        asRunConfig(config,output[0])
+
+
+rule prep_io_data:
+    input:
+        "../../../out/well_observed_trn_inputs.zarr",
+        "../../../out/well_observed_trn_targets.zarr",
+    output:
+        "{outdir}/nstates_{nstates}/rep_{rep}/prepped.npz"
```

Comment:
Does this mean that the `prepped.npz` file within each directory is duplicated? Would `prepped.npz` change based on the `wildcard.nstates` value?

Reply:
No. You are right. We could actually just have one "prepped.npz" file for all n_states and reps.